Documentation/Reference.md

```diff
@@ -341,3 +358,21 @@ A C language target is build with following flags in release mode:
 ### Depending on Apple Modules
 
 At this time there is no explicit support for depending on Foundation, AppKit, etc, though importing these modules should work if they are present in the proper system location. We will add explicit support for system dependencies in the future. Note that at this time the Package Manager has no support for iOS, watchOS, or tvOS platforms.
+
+## C language targets
+
+The C language module are laid out similar to Swift targets execept that the C langauge libraries should contain a directory named `include` to hold the public headers.  
+To allow a Swift module to import a C language module, add a [target dependency](#targets) in the manifest file. SwiftPM will automatically generate a modulemap for each C language library module for these 3 cases:
+
+* "include/Foo/Foo.h" exists and `Foo` is the only directory under include directory.  
+   Generates: `umbrella header "/path/to/include/foo/foo.h"`
+
+* "include/foo.h" exists and include contains no other directory.  
+   Generates: `umbrella header "/path/to/include/foo.h"`
+
+*  Otherwise in all other cases.  
+   Generates: `umbrella "path/to/include"
+
+In case of complicated `include` layouts, a custom `module.modulemap` can be provided inside `include`. SwiftPM will error out if it can not generate a modulemap w.r.t the above rules.
```

Comment:
No, it errors out because it might be a dependency to some Swift target, I guess we should find out and only generate the modulemap if it is actually a dependency...